components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -44,39 +56,93 @@ public void init() throws ArtifactSynchronizerException {
     public List<String> retrieveAllArtifacts(String label) throws ArtifactSynchronizerException {
         List<String> gatewayRuntimeArtifactsArray = new ArrayList<>();
         try {
-            List<ByteArrayInputStream> baip = apiMgtDAO.getAllGatewayPublishedAPIArtifacts(label);
-            for (ByteArrayInputStream byteStream :baip){
-                byte[] bytes = ByteStreams.toByteArray(byteStream);
-                String  gatewayRuntimeArtifacts = new String(bytes);
-                gatewayRuntimeArtifactsArray.add(gatewayRuntimeArtifacts);
+            String baseURL = APIConstants.HTTPS_PROTOCOL_URL_PREFIX + System.getProperty(APIConstants.KEYMANAGER_HOSTNAME) + ":" +
+                    System.getProperty(APIConstants.KEYMANAGER_PORT) + APIConstants.INTERNAL_WEB_APP_EP;
+            String endcodedgatewayLabel= URLEncoder.encode(label, APIConstants.DigestAuthConstants.CHARSET);
+            String path  = APIConstants.GatewayArtifactSynchronizer.GATEAY_SYNAPSE_ARTIFACTS
+                    + "?gatewayLabel="+ endcodedgatewayLabel;
+            String endpoint = baseURL + path;
+            HttpResponse httpResponse = invokeService(endpoint);
+            String responseString = EntityUtils.toString(httpResponse.getEntity(), APIConstants.DigestAuthConstants.CHARSET);
+            JSONObject artifactObject = new JSONObject(responseString);
+            JSONArray jArray = (JSONArray)artifactObject.get("list");
+            if (jArray != null) {
+                for (int i = 0; i < jArray.length(); i++) {
+                    gatewayRuntimeArtifactsArray.add(jArray.get(i).toString());
+                }
             }
-            if (log.isDebugEnabled()){
-                log.debug("Successfully retrieved Artifacts from DB");
+            return gatewayRuntimeArtifactsArray;
+        } catch (IOException e) {
+            String msg = "Error while executing the http client " ;
+            log.error(msg, e);
+            throw new ArtifactSynchronizerException(msg, e);
+        }
+    }
+
+    private HttpResponse invokeService(String endpoint) throws IOException, ArtifactSynchronizerException {
+        HttpGet method = new HttpGet(endpoint);
+        URL synapseGetURL = new URL(endpoint);
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance()
+                .getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String username = config.getFirstProperty(APIConstants.API_KEY_VALIDATOR_USERNAME);
+        String password = config.getFirstProperty(APIConstants.API_KEY_VALIDATOR_PASSWORD);
+        byte[] credentials = Base64.encodeBase64((username + ":" + password).getBytes
+                (APIConstants.DigestAuthConstants.CHARSET));
+        int synapsePort = synapseGetURL .getPort();
+        String synapseProtocol = synapseGetURL .getProtocol();
+        method.setHeader("Authorization", "Basic " + new String(credentials,
+                APIConstants.DigestAuthConstants.CHARSET));
```

Comment:
Can't we use the same line without exceeding the row width?

Reply:
Nope . It is exceeding the limit 
![Screenshot from 2020-06-26 00-30-51](https://user-images.githubusercontent.com/60866295/85782624-53371780-b744-11ea-9c34-0c3b6220dc98.png)